### ISSUE TITLE
Don't skip tests related to symlinks on windows

### ISSUE THREAD

**Issue by imsodin** [syncthing/syncthing#4887]
As per https://github.com/syncthing/syncthing/pull/4886#issuecomment-382076308:

> [...] So with this in mind, maybe we can make an attempt to fix the following symlink related tests that are skipped on Windows (maybe not all of them symlink related, but clearly many):
> 
> <img width="364" alt="screen shot 2018-04-17 at 19 28 03" src="https://user-images.githubusercontent.com/125426/38886217-93e0045e-4275-11e8-99c2-13f45c3658e9.png">

`osutil.DebugSymlinkForTestsOnly` is the magic that makes it possible. For fs tests we'd need to split the osutil package to prevent cyclic imports.

**Comment by AudriusButkevicius** [comment]
DebugSymlinkForTestsOnly clearly needs to live in the fs package.

**Comment by calmh** [comment]
Yeah. And I guess the relevant tests are those that depend on us functioning in an environment where the user might have created symlinks (e.g., TestWatchSymlinkedRoot looks relevant) and not those that test our handling of pulling symlinks etc (which would obviously fail on Windows).

**Comment by imsodin** [comment]
Fixing these tests might now be as easy as waiting for Go1.11 and not skipping them on windows: https://github.com/golang/go/issues/22874#event-1583426090
